### PR TITLE
Add population comparison to QuestionnaireResponse

### PR DESCRIPTION
**Pull request by alectenharmsel**
* When a questionnaire is finished, the population comparison is added
* Added a couple miscellaneous deployment fixes too
* Added byebug and dotenv for ruby development

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -21,9 +21,9 @@ services:
     depends_on:
       - db
     environment:
-      - PROMIS_API_USERNAME
-      - PROMIS_API_PASSWORD
-      - DYNAMO_LOCATION
+      PROMIS_API_USERNAME: "${PROMIS_API_USERNAME}"
```
Comment (kcperkhealth) [inline docker-compose.yml]: Isn't the name of our app **PRISM**?  ...

File: api/app/models/patient.rb [hunk 2]
```diff
@@ -93,6 +93,16 @@ def gender_is_valid
       errors.add(:gender, 'is not valid code. Valid codes are: male, female, other, unknown')
     end
 
+    def given_name
+      return nil if name.nil? || name.empty?
```
[the remark this answers is not in the document]
Comment (alectenharmsel) [reply api/app/models/patient.rb]: `blank?` is a Rails extension, so unfortunately no.

File: api/app/models/questionnaire_response.rb [hunk 3]
```diff
@@ -50,7 +51,9 @@ def reload
   #
   # If an item with the same link ID already exists, it's updated.
   def add_item(item_data)
-    existing = items&.index { |it| it.linkId == item_data[:linkId] }
+    existing = items&.index do |it|
+      it.linkId == item_data.with_indifferent_access[:linkId]
+    end
 
     if existing
```
[the remark this answers is not in the document]
Comment (alectenharmsel) [reply api/app/models/questionnaire_response.rb]: yes - all non-`nil` numbers evaluate to `true`

File: api/app/services/population.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                          elsif age < 45
+                            ['35 <= age < 45', COL_AGES[:lt_45]]
+                          elsif age < 55
+                            ['45 <= age < 55', COL_AGES[:lt_55]]
+                          elsif age < 65
+                            ['55 <= age < 65', COL_AGES[:lt_65]]
+                          elsif age < 75
+                            ['65 <= age < 75', COL_AGES[:lt_75]]
+                          else
+                            ['age > 75', COL_AGES[:gte_75]]
```
[the remark this answers is not in the document]
Comment (alectenharmsel) [reply api/app/services/population.rb]: I'd like to keep it as it is. The only code that can make it into the `else` is either `age >= 75`.

File: docker-compose.yml (hunk 1 above)
Comment (alectenharmsel) [reply docker-compose.yml]: Yep. Promis is an API we use to get questions for users to answer.

File: api/app/models/patient.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (kcperkhealth) [reply api/app/models/patient.rb]: I didn't think it is a sinatra app.

File: api/app/services/population.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (alectenharmsel) [reply api/app/services/population.rb]: description is fixed now. my bad.

**Pull request merged**